### PR TITLE
Adds typings for es6-collections and es6-shim libraries

### PR DESCRIPTION
**Pull request by rbuckton**
Adds typings for the following projects:
- [es6-collections](https://github.com/WebReflection/es6-collections/)
- [es6-shim](https://github.com/paulmillr/es6-shim)

### CODE REVIEW

**Comment by vvakame** [comment]
*es6-collections/es6-collections.d.ts*

check list

* [ ] is collect [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)?
  * https://www.npmjs.com/package/es6-collections
  * http://bower.io/search/?q=es6-collections
  * others?
* [X] has a [test file](http://definitelytyped.org/guides/contributing.html#tests)? (es6-collections/es6-collections-tests.ts or others)
* [ ] pass the Travic-CI test?

---

*es6-shim/es6-shim.d.ts*

check list

* [ ] is collect [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)?
  * https://www.npmjs.com/package/es6-shim
  * http://bower.io/search/?q=es6-shim
  * others?
* [X] has a [test file](http://definitelytyped.org/guides/contributing.html#tests)? (es6-shim/es6-shim-tests.ts or others)
* [ ] pass the Travic-CI test?

File: es6-collections/es6-collections.d.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
+// Type definitions for es6-collections v0.5.1
+// Project: https://github.com/WebReflection/es6-collections/
+// Definitions by: Ron Buckton <http://github.com/rbuckton>
+// Definitions: https://github.com/borisyankov/DefinitelyTyped
+
+/* *****************************************************************************
+Copyright (c) Microsoft Corporation. All rights reserved.
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use
```
Comment (vvakame) [inline es6-collections/es6-collections.d.ts]: This repository under the MIT license. https://github.com/borisyankov/DefinitelyTyped/blob/master/LICENSE
Where is this definition file from?

Comment (rbuckton) [inline es6-collections/es6-collections.d.ts]: I authored it to support the work I do on TypeScript. The current license header is the one we use on TypeScript. There is precedent for this, as the jQuery.d.ts that is currently part of DefinitelyTyped shares the same license header. Please let me know if this is an issue.

Comment (vvakame) [inline es6-collections/es6-collections.d.ts]: i see.

**Comment by vvakame** [comment]
@rbuckton it merged. thanks!

**Pull request merged**